### PR TITLE
Added 32-bit support.

### PR DESCRIPTION
**Pull request by Esption**
For issue #347

Main two things are causing it to pull and output both the 32 and 64 bit versions of float/double in the codegen. Unfortunately can't check `target_pointer_width` in codegen (it only sees your host width, not target) so that check has to be after codegen.

Also, need to change the align on Opaque for pointer casting.  AFAIK this should be complete besides tests.

### CODE REVIEW

File: godot-ffi/src/opaque.rs [hunk 1]
```diff
@@ -11,7 +11,8 @@
 ///
 /// Note: due to `align(8)` and not `packed` repr, this type may be bigger than `N` bytes
```
Comment (Bromeon) [inline godot-ffi/src/opaque.rs:12]: Comment would need to be updated:
```suggestion
/// Note: due to `align(4)`/ `align(8)` and not `packed` repr, this type may be bigger than `N` bytes
```

**Comment by GodotRust** [comment]
API docs are being generated and will be shortly available at: https://godot-rust.github.io/docs/gdext/pr-361

**Review by Bromeon — COMMENTED** [review]
Thanks a lot!

> Unfortunately can't check `target_pointer_width` in codegen (it only sees your host width, not target) so that check has to be after codegen.

Do you know why that is? `godot-codegen` is a normal Rust crate... the only special thing is that it's a build-dependency, not a regular one.

Also, which rustc target did you use? I wonder how we can best try to reproduce it in CI... 🤔

**Comment by Esption** [comment]
https://github.com/rust-lang/rust/issues/42587

It's just a byproduct of how it works.  It's unfortunate but the only other way is adding a feature cfg but that would feel clunky on the user side.

File: godot-codegen/src/api_parser.rs [hunk 2]
```diff
@@ -241,15 +241,17 @@ impl MethodReturn {
...
-    #[cfg(not(feature = "double-precision"))]
-    let build_config = "float_64"; // TODO infer this
-
+    let build_config: [&'static str; 2] = {
+        if cfg!(feature = "double-precision") {
+            ["double_32", "double_64"]
+        } else {
+            ["float_32", "float_64"]
+        }
+    };
```
Comment (Bromeon) [inline godot-codegen/src/api_parser.rs:254]: Thanks for explanation!

Could you maybe add that link with a short explanation above this snippet? That way, the design decision is documented and people don't wonder why 32/64 is not already determined at this point (like me) 🙂

**Comment by Bromeon** [comment]
rustfmt would need to be fixed as well, otherwise looks good! Please amend your existing commit if possible.

I still need to think about how to run 32-bit config in CI. What's the name of your target (the triple format)?

**Comment by Esption** [comment]
Forgot to do the cargo fmt, so yeah oops.  Also, I don't see any clippy things from anything I touched too tho there seem to be some floating around.

As for targets I'm just on windows atm so all I can really test is native windows or if it compiles in WSL. So, `i686-pc-windows-msvc` and `i686-unknown-linux-gnu`. It looks like the CI uses Ubuntu, which I don't believe installs 32-bit headers by default.  Likely you'd need to install `gcc-multilib` and possibly `g++-multilib`.

**Comment by Bromeon** [comment]
Thanks! Small request:

> Please amend your existing commit if possible.

Could you squash your commits into one?

**Comment by Bromeon** [comment]
Thank you! 🚀

**Pull request merged**